src/toil/batchSystems/mesos/batchSystem.py

```diff
@@ -388,9 +393,47 @@ def resourceOffers(self, driver, offers):
                 self._updateStateToRunning(offer, task)
                 log.info("...launching Mesos task %s" % task.task_id.value)
 
-            if len(tasks) == 0:
-                log.debug("Offer %s not large enough to run any tasks. Required: %s Offered: %s"
-                          % (offer.id.value, job_types[-1], (self.__mbToBytes(offerMem), offerCores, self.__mbToBytes(offerStor))))
+            # if we have no tasks scheduled to run, and we have no running jobs, we may never make
+            # progress due to a scheduler request issue
+            #
+            # the situation is not as dire if we're not the only framework running on mesos,
+            # but if this toil DAG is the only DAG in the system, we will hang here, and the
+            # user will wonder why forever
+            if (len(tasks) == 0 and
+                len(self.getRunningBatchJobIDs()) == 0):
+                
+                task_explanations = []
+
+                for job_type in job_types:
+
+                    resource_limits = []
+
+                    # which resources did we not have enough of?
+                    if offerCores < job_type.cores:
+                        resource_limits.append("insufficient cores (%d offered, %d requested)" % (offerCores, job_type.cores))
+                    if offerMem < self.__bytesToMB(job_type.memory):
+                        resource_limits.append("insufficient memory (%d MB offered, %d MB requested)" % (offerMem, self.__bytesToMB(job_type.memory)))
+                    if offerStor < self.__bytesToMB(job_type.disk):
+                        resource_limits.append("insufficient disk space (%d MB offered, %d MB requested)" % (offerStor, self.__bytesToMB(job_type.disk)))
+
+                    # log the resource request as well as the explanation
+                    task_explanations.append("%s: %s" % (job_type, ", ".join(resource_limits)))
+
+                log.warn("Although no jobs are running, offer %s with memory %dMB, %d cores, and disk %dMB is not large enough to run any tasks." % 
+                          (offer.id.value, offerMem, offerCores, offerStor))
+                log.warn("Pending job types are:\n%s" % '\n'.join(task_explanations))
+
+            else:
```

Comment:
Might want to nest an `if` instead of evaluating len(tasks) twice. Also `if len(foo) ==  0:` is equivalent to `if not foo:`. That plus inverting the condition and swapping the `if` and `else` bodies will usually make for more readable code.